lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -3211,6 +3211,52 @@ void BoundInterpreterFunction::fwdSparseLengthsWeightedSumGradInst(
   }
 }
 
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdSparseLengthsWeightedSumOffsetsInstFloatImpl(
+    const SparseLengthsWeightedSumOffsetsInst *I) {
+  staticAssertFloatingPointType(ElemTy);
+
+  auto out = getTensor(I->getDest());
+  auto data = getTensor(I->getData());
+  auto weights = getTensor(I->getWeights());
+  auto indices = getTensor(I->getIndices());
+  auto offsets = getTensor(I->getOffsets());
+
+  out->zero();
+
+  auto IH = indices->getHandle<int64_t>();
+  auto OFFH = offsets->getHandle<int64_t>();
+
+  size_t segments = offsets->dims()[0];
+  size_t totalLength = indices->dims()[0];
+
+  size_t lineSize = data->size() / data->dims()[0];
+
+  auto DH = data->getHandle<ElemTy>();
+  auto WH = weights->getHandle<ElemTy>();
+  auto OH = out->getHandle<ElemTy>();
+
+  size_t curIdx = 0;
+  for (size_t i = 0; i < segments; i++) {
+    size_t start = OFFH.raw(i);
+    size_t end = i == segments - 1 ? totalLength : OFFH.raw(i + 1);
```

Comment:
@jackm321 I meant this line

Reply:
Ah ok thanks, yeah this is the thing that won't work. Because irl totalLength won't actually be the size of the "real" inputs.

Reply:
Ok. I guess we'll just depend on weights padding for now.